docs/bis/ec/ec-views.md

```diff
@@ -0,0 +1,78 @@
+- Views can be applied only to the ECEntityClass or ECRelationshipClass.
+- For relationship classes, a different custom attribute `ECDbMap.ImplicitView` can be used instead, which will make the runtime go and check both sides for a navigation property which matches the relationship. If one is found, a view for the relationship is automatically generated.
+- Support for relationship views other than ImplicitView is currently limited. Using it for linktable relationships is not currently allowed.
+- Any metadata like property category or kind of quantity is taken from the property definitions. Metadata that may come from the view query is overridden by these.
+
+Example of a schema using a view:
+
+```xml
+<ECSchema schemaName="TestSchema" alias="ts" version="1.0.0" xmlns="http://www.bentley.com/schemas/Bentley.ECXML.3.2">
+    <ECSchemaReference name='ECDbMap' version='02.00.03' alias='ecdbmap' />
+    <ECEntityClass typeName="JsonObject">
+        <ECProperty propertyName="json" typeName="string" extendedTypeName="Json" />
+    </ECEntityClass>
+    <ECEntityClass typeName="Pipe" modifier="Abstract">
+        <ECCustomAttributes>
+            <View xmlns="ECDbMap.02.00.03">
+                <Query>
+                    SELECT
+                        jo.ECInstanceId,
+                        CAST(json_extract(jo.json, '$.diameter') AS INTEGER) [Diameter],
+                        CAST(json_extract(jo.json, '$.length') AS INTEGER) [Length],
+                        json_extract(jo.json, '$.material') [Material]
+                    FROM ts.JsonObject jo
+                    WHERE json_extract(jo.json, '$.type') = 'pipe'
+                </Query>
+            </View>
+        </ECCustomAttributes>
+        <ECProperty propertyName="Diameter" typeName="int" />
+        <ECProperty propertyName="Length"  typeName="int"/>
+        <ECProperty propertyName="Material" typeName="string" />
+    </ECEntityClass>
+</ECSchema>
+```
+
+`JsonObject` could be filled with this data:
+
+```
+{"type": "pipe", "diameter": 10, "length": 100, "material": "steel"}
+{"type": "pipe", "diameter": 15, "length": 200, "material": "copper"}
+{"type": "pipe", "diameter": 20, "length": 150, "material": "plastic"}
+{"type": "cable", "diameter": 5, "length": 500, "material": "copper","type": "coaxial"}
+{"type": "cable", "diameter": 2, "length": 1000, "material": "fiber optic","type": "single-mode"}
+{"type": "cable", "diameter": 3, "length": 750, "material": "aluminum","type": "twisted pair"}
+```
+
+Running this query:
+
+```sql
+SELECT Length, Diameter, Material FROM ts.Pipe
+```
+
+Will return this result:
+
+```
+Length   |Diameter |Material
+----------------------------------------
+100      |10       |steel
+200      |15       |copper
+150      |20       |plastic
+```
+
+A view may contain navigation properties, for these, the query can use the `navigation_value`function provided by ECSQL [ECSqlFunctions](./../../learning/ECSqlReference/ECSqlFunctions.md)
```

Comment:
it would be good to have another example that shows navigation properties

Reply:
I added a new example, check ECSqlReference/View.md